dev/benchmarks/macrobenchmarks/lib/src/web/bench_complex_layout.dart

```diff
@@ -0,0 +1,716 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart' show timeDilation;
+import 'recorder.dart';
+
+/// Measures how expensive it is to construct deep tree of widgets.
+class BenchBuildComplexLayout extends WidgetRecorder {
```

Comment:
What are the noise levels like for this benchmark? Curious, because this is probably the most macro benchmark we'll have.